ingest-app/src/cmr/ingest/api/generic_documents.clj

```diff
@@ -56,6 +62,20 @@
         (js-validater/validate-json schema-obj raw-json))
       ["Schema not found"])))
 
+(defn get-sub-concept-type-concept-id-prefix
+  "There are many concept types within generics. Read in the concept-id prefix for this specific one."
+  [spec-key version]
+  (if-some [index-url (jio/resource (format "generics/%s/v%s/index.json"  
+                                            (name spec-key)  
+                                            version))]
+    (let [index-file-str (slurp index-url)
+          index-file (json/parse-string index-file-str true)
+          index-sub-concept (:SubConceptType index-file)]
```

Comment:
if the schema does not require this field, then a default of 'X-' should be assumed.

Reply:
It is that is what the last line does: 
(:generic (set/map-invert cmr.common.concepts/concept-prefix->concept-type))))